protocol-units/settlement/mcr/contracts/src/staking/MovementStaking.sol

```diff
@@ -132,15 +135,15 @@ contract MovementStaking is
         address delegator,
         uint256 amount
     ) internal {
+        uint256 stake = epochStakesByDomain[domain][epoch][custodian][attester][
+            delegator
+        ];
+        stake -= amount; // okay to let this error if it underflows
         epochStakesByDomain[domain][epoch][custodian][attester][
             delegator
-        ] -= amount;
+        ] = stake;
```

Comment:
You could still use a storage reference here? This change only replaces one 5-step lookup for another.

Reply:
This is outdated, but I was doing this to save the extra stake lookups later on. You do also in theory save one instruction in the bytecode here.